iceoryx_posh/include/iceoryx_posh/internal/mepoo/typed_mem_pool.inl

```diff
@@ -51,8 +51,11 @@ inline cxx::expected<ChunkManagement*, TypedMemPoolError> TypedMemPool<T>::acqui
         return cxx::error<TypedMemPoolError>(TypedMemPoolError::FatalErrorReachedInconsistentState);
     }
 
-    new (chunkHeader) ChunkHeader(
-        m_memPool.getChunkSize(), sizeof(T), alignof(T), CHUNK_NO_CUSTOM_HEADER_SIZE, CHUNK_NO_CUSTOM_HEADER_ALIGNMENT);
+    auto chunkSettingsResult = mepoo::ChunkSettings::create(sizeof(T), alignof(T));
+    // this is safe since we use correct values for size and alignment
```

Comment:
Then please add `cxx::Expects` for the unlikely case that we change the requirements for correct size and alignment later and this goes south.

Reply:
I had it first, but decided to remove it since `sizeof` and `alignof` is used and I wouldn't be able to test the `cxx::Expects`